src/main/java/seedu/address/model/module/Module.java

```diff
@@ -37,6 +40,17 @@ public boolean isSameModuleCode(Module otherModule) {
         return otherModule == this || otherModule.moduleCode.equals(this.moduleCode);
     }
 
+    /**
+     * Creates and returns a {@code Module} with the details of {@code this}
+     * edited with {@code editModuleDescriptor}.
+     */
+    public Module edit(EditModuleCommand.EditModuleDescriptor editModuleDescriptor) {
+        requireNonNull(editModuleDescriptor);
+
+        ModuleCode updatedModuleCode = editModuleDescriptor.getModuleCode().orElse(this.moduleCode);
```

Comment:
Just want to clarify if the intention is for nullpointerexception to be thrown in next line after this is passed as param into
Module.

Reply:
Its similar to lixin's edit method in the task class. Yes it is for null pointer exception to be thrown. Under the edit module method, a non-null editModuleDescriptor object should be passed into this method.